### PR TITLE
add bin_shrink_image_filter

### LINKED ISSUE #8: Implement bin_shrink_image_filter

### PR DESCRIPTION
**Pull request by grlee77**
closes #8

### CODE REVIEW

File: itk_cucim/filtering/image_grid.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+from cucim.skimage.transform import downscale_local_mean
+
+
+def cucim_bin_shrink_image_filter(*args, **kwargs):
+    input_image = args[0]
+    ref_kwargs = kwargs.copy()
+    if 'shrink_factor' in ref_kwargs:
+        ref_kwargs['ShrinkFactor'] = ref_kwargs.pop('shrink_factor')
+    if 'shrink_factors' in ref_kwargs:
+        ref_kwargs['ShrinkFactors'] = ref_kwargs.pop('shrink_factors')
```
Comment (grlee77) [inline itk_cucim/filtering/image_grid.py:14]: Unlike on previous PRs, I had to manually swap the case of this here. Is that expected or should there be a change on the ITK side?

`itk.bin_shrink_image_filter` does work with the lowercase `shrink_factors`

Comment (thewtex) [reply itk_cucim/filtering/image_grid.py:14]: That's interesting. It should work with both.

File: test/filtering/test_image_grid.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    def setup_class(self):
+        data = Path(__file__).absolute().parent.parent / "input" / "head_mr.mha"
+        # uint8 data
+        image_u8 = itk.imread(data)
+        self.image = image_u8
+
+        Caster = itk.CastImageFilter[itk.itkImagePython.itkImageUC3,
+                                     itk.itkImagePython.itkImageF3].New()
+        # float32 data
+        self.image_f32 = Caster(image_u8)
```
Comment (thewtex) [inline test/filtering/test_image_grid.py]: Does,

```
self.image_f32 = image_u8.astype(np.float32)
```

work here?

**Review by thewtex — APPROVED** [review]
@grlee77 beautiful! :1st_place_medal: 

Minor style suggestion inline.

File: test/filtering/test_image_grid.py (hunk 2 above)
Comment (grlee77) [reply test/filtering/test_image_grid.py]: Yes, that is much simpler. I didn't realize the `itkImagePython` class had an `astype` method (I see there is a `dtype` attribute as well).

**Comment by thewtex** [comment]
Catching up on this it looks. Merging to keep moving forward.

**Pull request merged**